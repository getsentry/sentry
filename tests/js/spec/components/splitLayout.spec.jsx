import React from 'react';
import {shallow} from 'sentry-test/enzyme';
import toJson from 'enzyme-to-json';
import SplitLayout from 'app/components/splitLayout';

describe('SplitLayout', function() {
  it('renders with one child', function() {
    const component = shallow(
      <SplitLayout>
        <div>child</div>
      </SplitLayout>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with multiple children', function() {
    const component = shallow(
      <SplitLayout>
        <div>child #1</div>
        <div>child #2</div>
        <div>child #3</div>
      </SplitLayout>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with responsive property', function() {
    const component = shallow(
      <SplitLayout responsive>
        <div>child #1</div>
        <div>child #2</div>
        <div>child #3</div>
      </SplitLayout>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with a separation width', function() {
    const component = shallow(
      <SplitLayout responsive splitWidth={5}>
        <div>child #1</div>
        <div>child #2</div>
        <div>child #3</div>
      </SplitLayout>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
