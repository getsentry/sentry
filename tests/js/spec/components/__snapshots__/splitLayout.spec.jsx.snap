// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitLayout renders with a separation width 1`] = `
<SpreadLayout
  className="split-layout allow-responsive"
>
  <div
    className="split-layout-child"
    key=".0"
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    child #1
  </div>
  <div
    className="split-layout-child"
    key=".1"
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    child #2
  </div>
  <div
    className="split-layout-child"
    key=".2"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #3
  </div>
</SpreadLayout>
`;

exports[`SplitLayout renders with multiple children 1`] = `
<SpreadLayout
  className="split-layout"
>
  <div
    className="split-layout-child"
    key=".0"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #1
  </div>
  <div
    className="split-layout-child"
    key=".1"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #2
  </div>
  <div
    className="split-layout-child"
    key=".2"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #3
  </div>
</SpreadLayout>
`;

exports[`SplitLayout renders with one child 1`] = `
<SpreadLayout
  className="split-layout"
>
  <div
    className="split-layout-child"
    key=".0"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child
  </div>
</SpreadLayout>
`;

exports[`SplitLayout renders with responsive property 1`] = `
<SpreadLayout
  className="split-layout allow-responsive"
>
  <div
    className="split-layout-child"
    key=".0"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #1
  </div>
  <div
    className="split-layout-child"
    key=".1"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #2
  </div>
  <div
    className="split-layout-child"
    key=".2"
    style={
      Object {
        "marginRight": undefined,
      }
    }
  >
    child #3
  </div>
</SpreadLayout>
`;
